Clean up provided dependencies

mincong-h opened the pull request:
Issue #72.

- `org.jboss.spec.javax.batch` is replaced by `javax.batch`, which I believe more standard for users.
- `javax.enterprise` was here because this was written in https://jberet.gitbooks.io/jberet-user-guide/content/set_up_jberet/ But I think this page only explains how JBeret works behind the screen. It doesn't mean that we need to depends on `javax.entreprise:cdi-api` directly. So this dependency can be deleted safely.

As for `javax.inject`, I don't know what to do... In the spec page 4,

> This specification describes the job specification language, Java programming model, ... **Additionally, it is designed to work with dependency injection (DI) containers without prescribing a particular DI implementation.**

It seems that spec does not depends on `javax.inject` directly, inside, this spec should work with both CDI and DI. Therefore, if `javax.inject` is deleted right now, the code breaks immediately. I think we can handle it in another PR.

mincong-h committed:
#72 clean up dependencies

yrodiere commented on core/pom.xml:
```diff
@@ -39,19 +40,14 @@
-            <scope>provided</scope>
+            <groupId>javax.batch</groupId>
+            <artifactId>javax.batch-api</artifactId>
+            <version>${javax.batch}</version>
```
To be more in line with what's done for other dependencies, I think we should:

 * remove the `version` tag from here
 * move the `javax.batch` property to the parent pom
 * remove the dependency management for `jboss-batch-api_1.0_spec` in the parent pom
 * add dependency management for `javax.batch-api` in the parent pom

yrodiere replied:
Why isn't this dependency marked as "provided" as `jboss-batch-api_1.0_spec` was?
I think it would make sense, especially given we used the "provided" scope for `javax.inject`.
Note that the scope of the dependency can be omitted here and moved up to the parent's dependency management (same for `javax.inject`).

yrodiere commented on core/pom.xml:
```diff
@@ -39,19 +40,14 @@
-        </dependency>
-        <dependency>
             <groupId>javax.inject</groupId>
             <artifactId>javax.inject</artifactId>
```
Ok, you're right, it seems there is no such thing as transitive provided dependencies, even just when compiling. Which makes sense, I guess. So yes, we need this explicit "provided" dependency to make the build pass.

yrodiere reviewed (changes requested):
Thanks for submitting this PR! I have a few comments below.

mincong-h committed:
Clean up dependencies
- "jboss-batch-api_1.0_spec" is replaced by "javax.batch-api"
- "javax.batch-api" has scope provided, implementation depends on user's choice / platform
- "javax.batch-api" version is managed in the parent POM

mincong-h commented:
Hi, @yrodiere. I cleaned up the dependencies as we've discussed:

- `jboss-batch-api_1.0_spec` is replaced by `javax.batch-api`
- `javax.batch-api` has scope provided, implementation depends on user's choice / platform
- `javax.batch-api` version is managed in the parent POM

However, I did a rebase on this branch, commits are messed up ... 😞

yrodiere commented:
> However, I did a rebase on this branch, commits are messed up ... :disappointed:

I don't know what happened exactly, but here are a few recommendations. They make my life easier, but you're obviously free to do as you want :)

 * Set up your git config to do rebases when you pull from upstream, instead of merging (it will avoid merge commits to appear without you asking anything): `git config --global pull.rebase true`. See [here](https://coderwall.com/p/7aymfa/please-oh-please-use-git-pull-rebase) or [here](http://gitready.com/advanced/2009/02/11/pull-with-rebase.html) for explanations.
 * When you rebase, use the `-i` option so you can see what commits will be rebased
 * When in doubt, use `git help rebase` to see exactly what version of the command you should use. I find the syntax rather confusing, but the documentation is great: it explains multiple use cases, each time with a schema of what you have before, what you want after, and the appropriate command to run in order to get this result.

I took the liberty to rebase this branch and force-push; everything should be fine now.

yrodiere committed:
i72 Manage the scope "provided" of APIs in the parent pom

yrodiere committed:
i72 Clean up some more dependencies

 * Manage the "provided" scope of APIs in the parent pom
 * Manage most dependency versions in the parent pom
 * Remove unneeded explicit dependencies (hibernate-entitymanager not
   needed since ORM 5.2, hibernate-jpa-2.1-api not needed since we
   depend on hibernate-core)

I left the explicit versioning of jberet/byteman in -core for now.

yrodiere commented:
@mincong-h I added two more commits to clean up some other dependencies. In my opinion it's okay to merge if the build passes. If you agree, maybe you can merge? You can try the "squash and merge" option on the GitHub interface (use the drop-down button next to the "merge pull request" button).

mincong-h commented:
Thanks @yrodiere ! I think I need much more practice to handle git. I set `git config --global pull.rebase true` as you suggested. For the rest, I'll read later :)
